File: study/그리디/02-큰 수의 법칙.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const n = parseInt(input[0][0]);
const m = parseInt(input[0][2]);
const k = parseInt(input[0][4]);
const array = input[1].split(' ').map((el) => parseInt(el));

array.sort((a, b) => b - a);

const fisrt = array[0];
const second = array[1];

const firstNum = k * Math.trunc(m / k);
const secondNum = m % k;

const answer = fisrt * firstNum + second * secondNum;
console.log(answer);

File: solving/나동빈-이것이코딩테스트이다/그리디/02-큰 수의 법칙.py
import sys
getInput = sys.stdin.readline

n, m, k = map(int, getInput().rstrip().split()) # 배열의 크기, 숫자가 더해지는 횟수, 연속으로 더해질 수 있는 횟수
array = list(map(int, getInput().rstrip().split()))

array.sort(reverse=True)

first = array[0]
second = array[1]

first_num = k * (m // k)
second_num = m % k

answer = first * first_num + second * second_num
print(answer)

